core/src/main/java/org/stellar/anchor/sep10/JwtService.java

```diff
@@ -46,6 +48,16 @@ public JwtToken decode(String cipher) {
     JwtParser jwtParser = Jwts.parser();
     jwtParser.setSigningKey(jwtKey);
     Jwt jwt = jwtParser.parse(cipher);
+    Header header = jwt.getHeader();
+    if (!(header instanceof DefaultJwsHeader)) {
+      // This should not happen
+      throw new IllegalArgumentException("Bad token");
+    }
+    DefaultJwsHeader defaultHeader = (DefaultJwsHeader) header;
+    if (!defaultHeader.getAlgorithm().equals(SignatureAlgorithm.HS256.getValue())) {
+      // Not signed by the JWTService.
+      throw new IllegalArgumentException("Bad token");
```

Comment:
💡 Suggest distinguishing these exceptions with something like the following:
```suggestion
      throw new IllegalArgumentException("Unrecognized algorithm");
```

Reply:
This is purposely hidden not to reveal the algorithm that we are using for the attackers. 

Probably not too meaningful for they can tell from the signature. But just some extra precausion.

Reply:
The exception details make it up into the API response? We should probably revisit that, as exception details can contain information we don't want to leak in APIs.

Reply:
Yes, the exception description does bubble up to the exception handler unless we catch and wrap. The default exception handler does include the reason of the failure. So, we pay extra attention to the security related messages. 

I think this is probably ok for most API errors. But for security, it would help to have an extra pair of eyes to see if we miss anything.

Reply:
In this specific case the exception is caught and 403 Forbidden is returned which is good in terms of security.
I think even the generic exception handler should not return the exception message. I'll open an issue to discuss this.